app/src/main/kotlin/com/wire/android/ui/calling/controlButtons/JoinButton.kt

```diff
@@ -23,8 +25,10 @@ fun JoinButton(
     minHeight: Dp = MaterialTheme.wireDimensions.buttonMinSize.height,
     minWidth: Dp = MaterialTheme.wireDimensions.buttonMinSize.width
 ) {
+    val audioPermissionCheck = AudioBluetoothPermissionCheckFlow { buttonClick() }
+
     WirePrimaryButton(
-        onClick = buttonClick,
+        onClick = { audioPermissionCheck.launch() },
```

Comment:
```suggestion
        onClick = audioPermissionCheck::launch,
```

If possible we can use this, to avoid recompositions =)